Compose Dev Services do not work on Windows

### Describe the bug

Compose Dev Services do not work on Windows, execution ends with 
```
Caused by: io.smallrye.common.process.AbnormalExitException: Process exited abnormally (pid 6280) with exit code 125 with error output:
unknown shorthand flag: 'd' in -d
See 'docker --help'.
...
```

CC @ozangunalp 

The **root cause** is that docker-compose plugin is not found and thus docker execution fails. 


Me and @michalvavrik did several experiments on these environments:
 * Windows 10 with Docker Desktop
 * Windows Server with Docker CLI and Compose installed using Chocolatey
 
Our conclusion is that **some environment variables need to be propagated**.

For Michal (Windows Server with Docker CLI and Compose installed using Chocolatey) it was PROGRAMDATA to detect plugins in `C:\ProgramData\Docker\cli-plugins` as Chocolatey puts stuff there - https://github.com/geicht/chocolatey-packages/blob/master/docker-compose/tools/chocolateyInstall.ps1#L5

In my case (Windows 10 with Docker Desktop) it was PROGRAMFILES property so that docker-compose was detected.

I did changes in https://github.com/quarkusio/quarkus/blob/main/extensions/devservices/deployment/src/main/java/io/quarkus/devservices/deployment/compose/ComposeRunner.java#L103
The initial approach was `env.putAll(System.getenv());` to see that it makes difference and after that trying env properties one by one.
The one needed in my case was `env.put("PROGRAMFILES", System.getenv("PROGRAMFILES"));`

The right property is obviously end system dependent, there are also other env properties which may have influence on other systems - e.g. PROGRAMFILES(X86), COMMONPROGRAMFILES, COMMONPROGRAMFILES(X86).


To me it almost feels like `System.getenv()` should be reflected in the command execution.

Workaround in case you know the needed property: `quarkus.compose.devservices.env-variables.PROGRAMFILES=C:\\Program Files`
(Please note `\\` in the path)

### Expected behavior

Compose Dev Services work well on Windows

### Actual behavior

Compose Dev Services do not work on Windows, execution ends with 

### How to Reproduce?

 * Get application with jdbc-postgresql - https://code.quarkus.io/?e=jdbc-postgresql&e=rest
 * Add compose-devservices.yml as described in https://quarkus.io/guides/compose-dev-services#basic-example-postgresql-database
 * Run `mvn quarkuus:dev`

### Output of `uname -a` or `ver`

Windows

### Output of `java -version`

Java 21

### Quarkus version or git rev

Quarkus 3.25.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/devservices/deployment/src/main/java/io/quarkus/devservices/deployment/compose/ComposeRunner.java
+++ b/extensions/devservices/deployment/src/main/java/io/quarkus/devservices/deployment/compose/ComposeRunner.java
@@ -32,7 +32,7 @@ public class ComposeRunner {
     private final String identifier;
     private final String composeExecutable;
     private String cmd;
-    private Map<String, String> env;
+    private Map<String, String> additionalEnvVars;
     private List<String> profiles;
 
     public ComposeRunner(String composeExecutable, List<File> composeFiles, String projectName) {
@@ -40,8 +40,8 @@ public class ComposeRunner {
         this.composeFiles = composeFiles;
         this.identifier = projectName;
         this.cmd = "";
-        this.env = Collections.emptyMap();
-        this.profiles = Collections.emptyList();
+        this.additionalEnvVars = Map.of();
+        this.profiles = List.of();
     }
 
     /**
@@ -62,7 +62,7 @@ public class ComposeRunner {
      * @return this
      */
     public ComposeRunner withEnv(Map<String, String> env) {
-        this.env = Collections.unmodifiableMap(env);
+        this.additionalEnvVars = Collections.unmodifiableMap(env);
         return this;
     }
 
@@ -97,8 +97,9 @@ public class ComposeRunner {
         ProcessBuilder.newBuilder(composeExecutable)
                 .directory(pwd.toPath())
                 .arguments(cmd.split("\\s+"))
-                .environment(env)
                 .modifyEnvironment(env -> {
+                    env.putAll(additionalEnvVars);
+
                     env.put(PROJECT_NAME_ENV, identifier);
 
                     TransportConfig transportConfig = DockerClientFactory.instance().getTransportConfig();

